Framework/PythonInterface/test/python/plugins/algorithms/GSASIIRefineFitPeaksTest.py

```diff
@@ -1,67 +1,136 @@
 from __future__ import (absolute_import, division, print_function)
 
+import os
+import tempfile
 import unittest
-from mantid.simpleapi import *
+import mantid.simpleapi as mantid
 from mantid.api import *
 
-class GSASIIRefineFitPeaksTest(unittest.TestCase):
+
+class _GSASFinder(object):
     """
-    Very limited test, as executing this algorithm requires a modified
-    version of GSASII. At least it does some check that the algorithm is
-    registered succesfully and basic sanity checks.
+    Helper class for unit test - the algorithm can't run without a version of GSAS-II that includes the module
+    GSASIIscriptable (added April 2017)
     """
+    @staticmethod
+    def _find_directory_by_name(cur_dir_name, cur_dir_path, name_to_find, level, max_level):
+        """
+        Perform a depth-limited depth-first search to try and find a directory with a given name
+        """
+        if level == max_level:
+            return None
+
+        if cur_dir_name == name_to_find:
+            return cur_dir_path
+
+        list_dir = os.listdir(cur_dir_path)
+        for child in list_dir:
+            child_path = os.path.join(cur_dir_path, child)
+            if os.path.isdir(child_path):
+                try:
+                    path = _GSASFinder._find_directory_by_name(cur_dir_name=child, cur_dir_path=child_path,
+                                                               level=level + 1, name_to_find=name_to_find,
+                                                               max_level=max_level)
```

Comment:
Any reason we can't use `os.walk` here instead?

Reply:
I could be wrong, but don't think `os.walk` can be depth-limited (you can filter the results by depth, but you still have to go through the entire iterator), and on Linux at least does a depth-first search, so it'll fail on the build servers due to timeout

Reply:
In that case this is probably fine.